packages/teleport/src/NotificationContext.tsx

```diff
@@ -0,0 +1,30 @@
+import { createContext, useState } from 'react';
+
+// Notification Context
+type NotificationContextType = ReturnType<typeof makeNotificationContextValue>;
+
+const NotificationContext = createContext<NotificationContextType | null>(null);
+
+// Context Value
+const makeNotificationContextValue = () => {
+  const [notifications, setNotifications] = useState([]);
```

Comment:
```suggestion
  const [notifications, setNotifications] = useState<string[]>([]);
```

(disregard this if you make the architecture change below)